compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFuseConsumerIntoLoop.cpp

```diff
@@ -658,80 +36,103 @@ void AMDAIEFuseConsumerIntoLoopPass::runOnOperation() {
-                  "ContractionOpInterface \n");
-    return;
-  }
+    LoopLikeOpInterface loops = cast<LoopLikeOpInterface>(scfLoopOp);
+    auto loopBody = useSCFFor ? cast<scf::ForOp>(loops).getBody()
+                              : cast<scf::ForallOp>(loops).getBody();
+    // Search the compute op and its consumer slices.
+    linalg::LinalgOp linalgOp;
+    // TODO(avarma): All the following combination seems to be giving the same
+    //               results for an IR Walk :-
+    //               1. PostOrder + ReverseIterator.
+    //               2. PreOrder.
+    //               3. PreOrder + ReverseIterator.
+    //               Something while IR formation seems to be performing an ill
+    //               operation maybe?
+    loopBody->walk<WalkOrder::PostOrder, ReverseIterator>(
+        [&](linalg::LinalgOp op) {
+          if ((depth == 1 && linalg::isaContractionOpInterface(op)) ||
+              (depth == 2 && !linalg::isaContractionOpInterface(op))) {
+            linalgOp = op;
+            return WalkResult::interrupt();
+          }
+          return WalkResult::advance();
+        });
+
+    if (!linalgOp) {
+      LLVM_DEBUG(llvm::dbgs() << "----- Could not find any compute op \n");
+      return;
+    }
 
-  Value::user_range users = linalgOp->getResult(0).getUsers();
-  if (!llvm::hasSingleElement(users)) {
-    LLVM_DEBUG(llvm::dbgs()
-               << "----- Expected only one user of the compute op : "
-               << linalgOp << "\n");
-    return;
-  }
+    Value::user_range users = linalgOp->getResult(0).getUsers();
+    if (!llvm::hasSingleElement(users)) {
+      LLVM_DEBUG(llvm::dbgs()
+                 << "----- Expected only one user of the compute op : "
+                 << linalgOp << "\n");
+      return;
+    }
 
-  Operation *terminatorStoreOp = *(users.begin());
-  if (!(isa<tensor::InsertSliceOp, tensor::ParallelInsertSliceOp>(
-          terminatorStoreOp))) {
-    LLVM_DEBUG(llvm::dbgs()
-               << "----- Expected either tensor.insert_slice OR "
-                  "tensor.parallel_insert_slice to be the only user of the "
-                  "compute op : "
-               << linalgOp << "\n");
-    return;
-  }
+    Operation *terminatorStoreOp = *(users.begin());
+    if (!(isa<tensor::InsertSliceOp, tensor::ParallelInsertSliceOp>(
+            terminatorStoreOp))) {
+      LLVM_DEBUG(llvm::dbgs()
+                 << "----- Expected either tensor.insert_slice OR "
+                    "tensor.parallel_insert_slice to be the only user of the "
+                    "compute op : "
+                 << linalgOp << "\n");
+      return;
+    }
 
-  if (!terminatorStoreOp) {
-    LLVM_DEBUG(llvm::dbgs()
-               << "----- Consumer ops are already fused it seems \n");
-    return;
-  }
+    if (!terminatorStoreOp) {
+      LLVM_DEBUG(llvm::dbgs()
+                 << "----- Consumer ops are already fused it seems \n");
```

Comment:
Nit: Rephrase this message.

Reply:
I see you deleted this check. Is there a reason this is not needed anymore?

Reply:
This was a redundant check because in the earlier check I'm already making an exit if `terminatorStoreOp` is NOT `tensor.insert_slice/tensor.parallel_insert_slice`.

Hence I removed.